lib/rubocop/cop/rspec/implicit_subject.rb

```diff
@@ -66,32 +98,67 @@ def on_send(node)
         private
 
         def autocorrect(corrector, node)
-          replacement = 'expect(subject)'
           case node.method_name
+          when :expect
+            corrector.replace(node, 'is_expected')
+          when :is_expected
+            corrector.replace(node.location.selector, 'expect(subject)')
           when :should
-            replacement += '.to'
+            corrector.replace(node.location.selector, 'expect(subject).to')
           when :should_not
-            replacement += '.not_to'
+            corrector.replace(node.location.selector, 'expect(subject).not_to')
           end
-
-          corrector.replace(node.loc.selector, replacement)
         end
 
-        def valid_usage?(node)
-          example = node.ancestors.find { |parent| example?(parent) }
-          return false if example.nil?
-
-          example.method?(:its) || allowed_by_style?(example)
+        def message(_node)
+          case style
+          when :require_implicit
+            MSG_REQUIRE_IMPLICIT
+          else
+            MSG_REQUIRE_EXPLICIT
+          end
         end
 
-        def allowed_by_style?(example)
+        def invalid?(node)
```

Comment:
Wouldn't it be simpler, if this was tge `alloed_by_style?` implementation. Sure, it now need both the example and the node (for the require_implicit) style; but it also takes care of finding the example, and ignoring it if it's `its`. If the reason to go in this direction is to enforce the implicit usage in `its` - then this is not shown in the tests